sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -191,6 +191,8 @@ case class Expand(
     val sizeInBytes = child.statistics.sizeInBytes * projections.length
     Statistics(sizeInBytes = sizeInBytes)
   }
+
+  override def missingInput = super.missingInput.filter(_.name != VirtualColumn.groupingIdName)
```

Comment:
My point is we may also support other virtual columns like `INPUT__FILE__NAME` and `BLOCK__OFFSET__INSIDE__FILE` in the future. These virtual columns can be used in more general places like `Project` and `Filter`. That's why I think moving this to `QueryPlan` may make more sense.